src/map/hash_trie_map/mod.rs

```diff
@@ -1032,47 +1031,40 @@ enum IterStackElement<'a, K, V, P>
 where
     P: SharedPointerKind,
 {
-    Branch(Peekable<slice::Iter<'a, SharedPointer<Node<K, V, P>, P>>>),
-    LeafSingle(&'a EntryWithHash<K, V, P>),
-    LeafCollision(Peekable<list::Iter<'a, EntryWithHash<K, V, P>, P>>),
+    Branch(slice::Iter<'a, SharedPointer<Node<K, V, P>, P>>),
+    LeafSingle(Option<&'a SharedPointer<Entry<K, V>, P>>),
```

Comment:
I think you can remove the `Option<>` from here.

Reply:
It's needed in order to implement `next()` for `IterStackElement::LeafSingle`.

But there's a catch, `IterStackElement::LeafSingle` is never a thing because the trie root right now is never a leaf.
So I see two alternatives
1. leave it as is
2. make `IterStackElement::new` for `Node::Leaf(Bucket::Single)` panic, because it's really unreachable right now.

Reply:
You're right.  We could also replace the option with a `std::iter::Once` iterator, which is closer to what the `Option<>` is trying to represent.  What do you think?

Reply:
Sounds good. I'll update the PR to use std::iter::once.